Add initial support for bedrock

kujtimiihoxha opened the pull request:
To use claude bedrock 3.7 use the model 
```json
{
  "model": {
    "coder": "bedrock.claude-3.7-sonnet",
    "coderMaxTokens": 20000
  },
  "lsp": {
    "gopls": {
      "command": "gopls"
    }
  }
}
```

kujtimiihoxha commented:
@adamdottv this adds initial support for bedrock.

It uses the `AWS_REGION` or `DEFAULT_AWS_REGION` to prefix the model name (bedrock is weird)

Also termai does not do anything to setup your aws credentials so it assumes you are authenticated.

Also currently the bedrock provider does not use caching.

adamdottv commented:
@kujtimiihoxha i pulled latest and did a `go build`, but i'm not able to get it to send anything, when i type a new message and hit enter it just sits there (no debug log messages either)

kujtimiihoxha commented:
@adamdottv pull latest now and see what logs you see when you click 'L' I added some better logging support (to go back just hid backspace/delete). If you still don't see anything I can help debug.

adamdottv commented:
<img width="1878" alt="Screenshot 2025-04-09 at 12 39 09 PM" src="https://github.com/user-attachments/assets/17b5e3d3-9243-48da-bcf1-6d60530747f1" />

cc @kujtimiihoxha

kujtimiihoxha commented:
@adamdottv ok so I was able to reproduce this, the issue is the AWS session what I had to do to make it work is 
```
aws sso login --profile MY_PROFILE
export AWS_REGION=us-east-1
export AWS_PROFILE=MY_PROFILE

go run main.go
```

This works. Also pull the latest stuff I added the default region so the region env is not needed and also fixed some small things.

kujtimiihoxha commented:
@adamdottv make sure to also get the latest changes, I fixed a couple of things and made logs better, i also added the ability to cancel the message when you are focused on the editor just click `control+x` (if you have a better shortcut let me know 😂 )
